src/brod_group_coordinator.erl

```diff
@@ -911,44 +913,53 @@ get_committed_offsets(#state{ offset_commit_policy = commit_to_kafka_v2
         lists:foldl(
           fun(PartitionOffset, Acc) ->
             Partition = kpro:find(partition, PartitionOffset),
-            Offset = kpro:find(offset, PartitionOffset),
+            Offset0 = kpro:find(offset, PartitionOffset),
+            Metadata = kpro:find(metadata, PartitionOffset),
             EC = kpro:find(error_code, PartitionOffset),
             case EC =:= ?EC_UNKNOWN_TOPIC_OR_PARTITION of
               true ->
                 %% EC_UNKNOWN_TOPIC_OR_PARTITION is kept for version 0
                 Acc;
               false ->
                 ?ESCALATE_EC(EC),
+                Offset = maybe_upgrade_from_roundrobin_v1(Offset0, Metadata),
                 [{{Topic, Partition}, Offset} | Acc]
             end
           end, [], PartitionOffsets)
       end, TopicOffsets),
   lists:append(CommittedOffsets0).
 
 %% @private
--spec resolve_begin_offsets([TP], [{TP, brod:offset()}]) ->
+-spec resolve_begin_offsets([TP], [{TP, brod:offset()}], boolean()) ->
         brod:received_assignments()
           when TP :: {brod:topic(), brod:partition()}.
-resolve_begin_offsets([], _) -> [];
-resolve_begin_offsets([{Topic, Partition} | Rest], CommittedOffsets) ->
-  Offset =
+resolve_begin_offsets([], _CommittedOffsets, _IsConsumerManaged) -> [];
+resolve_begin_offsets([{Topic, Partition} | Rest], CommittedOffsets,
+                      IsConsumerManaged) ->
+  BeginOffset =
     case lists:keyfind({Topic, Partition}, 1, CommittedOffsets) of
-      {_, Offset_} when is_integer(Offset_) ->
-        Offset_;
+      {_, Offset} when IsConsumerManaged ->
+        %% roundrobin_v2 is only for kafka commits
+        %% for consumer managed offsets, it's still acked offsets
+        %% therefor we need to +1 as begin-offset
```

Comment:
therefore.

Also, how "consumer managed" flag is related to roundrobin_v2?

Reply:
`resolve_begin_offsets/3` is a function shared for `consumer_managed` offset commit policy and `commit_to_kafka_v2` policy.